dev/build_prevalidation.py

```diff
@@ -23,16 +23,48 @@
 from generate_bom import BomGenerator
 from refresh_source import Refresher
 
+from spinnaker.run import check_run_quick
+
 
 def __annotate_component(annotator, component):
   """Annotate the component's source but don't include it in the BOM.
+
+  Returns:
+    Tuple of ([VersionBump] Halyard version bump, [string] head commit hash)
   """
   annotator.path = component
   annotator.parse_git_tree()
-  annotator.tag_head()
+  version_bump = annotator.tag_head()
+  head_hash = annotator.get_head_commit()
   annotator.delete_unwanted_tags()
+  return (version_bump, head_hash)
+
+def __record_halyard_nightly_version(version_bump, head_hash, options):
+  """Record the version and commit hash at which Halyard was built in a bucket.
+
+  Assumes that gsutil is installed on the machine this script is run from.
+  """
+  bucket_uri = options.hal_nightly_bucket_uri
+  build_number = options.build_number or os.environ.get('BUILD_NUMBER', '')
+  local_bucket_name = os.path.basename(bucket_uri)
+  # Copy all the bucket contents to local (-r) and get rid of extra stuff (-d).
+  if not os.path.exists(local_bucket_name):
+    os.mkdir(local_bucket_name)
+  check_run_quick('gsutil rsync -r -d {remote_uri} {local_bucket}'
+                  .format(remote_uri=bucket_uri, local_bucket=local_bucket_name))
+  hal_version = version_bump.version_str.replace('version-', '')
+  new_hal_nightly_entry = ('{version}-{build}={commit}'
```

Comment:
Are you suggesting making this a nested list, or just keep the same structure but change `=` to `:`?